model/common/src/icon4py/model/common/utils/gt4py_field_allocation.py

```diff
@@ -5,13 +5,49 @@
 #
 # Please, refer to the LICENSE file in the root directory.
 # SPDX-License-Identifier: BSD-3-Clause
+import logging as log
 from typing import Optional
 
+import gt4py._core.definitions as gt_core_defs
 import gt4py.next as gtx
 from gt4py.next import backend
 
-from icon4py.model.common import type_alias as ta
-from icon4py.model.common.settings import xp
+from icon4py.model.common import dimension, type_alias as ta
+
+
+def is_cupy_device(backend: backend.Backend) -> bool:
+    cuda_device_types = (
+        gt_core_defs.DeviceType.CUDA,
+        gt_core_defs.DeviceType.CUDA_MANAGED,
+        gt_core_defs.DeviceType.ROCM,
+    )
+    if backend is not None:
+        return backend.allocator.__gt_device_type__ in cuda_device_types
```

Comment:
I am simply asking a question here. 
There is this function `is_field_allocator_for` in gt4py.next. I wonder if it is better to use that function  to check if it is the gpu backend. If not, why. For example, 
`if allocators.is_field_allocator_for(backend.allocator, core_defs.DeviceType.CUDA)`

Reply:
I remember that I had a look at this one, but then choose not to use it because we would need to call it for each GPU device type. 
```
if allocators.is_field_allocator_for(backend.allocator, core_defs.DeviceTyp.CUDA) or allocators.is_field_allocator_for(backend.allocator, core_defs.DeviceTyp.ROCM) or allocators.is_field_allocator_for(backend.allocator, core_defs.DeviceTyp.CUDA_MANAGED)
```
that looked very cumbersome to me.